### PR TITLE
Added KT estimation to fitting routines.

### PR DESCRIPTION
**Pull request by grahamrow**
Why use Julia's A\b when you can use np.matmul(np.linalg.pinv(A), b)?

### CODE REVIEW

File: src/auspex/analysis/fits.py [hunk 1]
```diff
@@ -1,9 +1,94 @@
 from scipy.optimize import curve_fit
 import numpy as np
 from numpy.fft import fft
+from scipy.linalg import svd, eig, inv, pinv
 from enum import Enum
 
-import matplotlib.pyplot as plt
+def hilbert(signal):
```
Comment (caryan) [inline src/auspex/analysis/fits.py]: why not just use scipy.signal.hilbert?

File: src/auspex/analysis/fits.py [hunk 2]
```diff
@@ -1,9 +1,94 @@
...
+    K = order
+    M = (N//2)-1
+    L = N-M+1
+    H = np.zeros((L, M), dtype=np.complex128)
+    for ct in range(M):
+        H[:,ct] = analytic_signal[ct:ct+L]
+
+    #Try and seperate the signal and noise subspace via the svd
+    U,S,V = svd(H, False)
+    V = V.T # not transposed/conjugated in numpy svd
```
Comment (caryan) [inline src/auspex/analysis/fits.py]: you're  just transposing again on line 46

File: src/auspex/analysis/fits.py [hunk 3]
```diff
@@ -1,9 +1,94 @@
...
+    #Compute Q with total least squares
+    #U_K1*Q = U_K2
+    U = svd(cleanedH, False)[0]
+    U_K = U[:,0:K]
+    tmpMat = np.hstack((U_K[:-1,:],U_K[1:,:]))
+    V = svd(tmpMat, False)[2].T.conj()
+    n = np.size(U_K,1)
+    V_AB = V[:n,n:]
+    V_BB = V[n:,n:]
+    Q = -np.matmul(V_AB,inv(V_BB)) # Julia conversion gotcha...
```
Comment (caryan) [inline src/auspex/analysis/fits.py]: May end up equivalent but you can avoid an `inv` with:

```
Q = np.linalg.lstsq(V_BB.conj().T, -V_AB.conj().T)[0].conj().T
```

File: src/auspex/analysis/fits.py [hunk 4]
```diff
@@ -1,9 +1,94 @@
...
+        sk = np.log(poles[ct])
+        freqs[ct] = np.imag(sk)/(2*np.pi*time_step)
+        Tcs[ct] = -1.0/np.real(sk)*time_step
+
+    #Refit the data to get the amplitude
+    A = np.zeros((N, K), dtype=np.complex128)
+    for ct in range(K):
+        A[:,ct] = np.power(poles[ct], range(0,N))
+
+    amps = np.matmul(pinv(A),cleanedData)
```
Comment (caryan) [inline src/auspex/analysis/fits.py]: similar `lstsq` here

```
amps = np.linalg.lstsq(A, cleaned_data)[0]
```

File: src/auspex/analysis/fits.py [hunk 5]
```diff
@@ -1,9 +1,94 @@
...
+    #Here we can iterate and modify the singular values
+    S_k = np.diag(S[:K])
+
+    #Estimate the variance from the rest of the singular values
+    varEst = (1/((M-K)*L)) * np.sum(S[K:]**2)
+    Sfilt = np.matmul(S_k**2 - L*varEst*np.eye(K), inv(S_k))
+    Hbar = np.matmul(np.matmul(U[:,:K], Sfilt), V[:,:K].T)
+
+    #Reconstruct the data from the averaged anti-diagonals
+    cleanedData = np.zeros(N, dtype=np.complex128)
```
Comment (caryan) [inline src/auspex/analysis/fits.py]: not going to take the chance to snakeify things?

**Comment by grahamrow** [comment]
The timing you mentioned is about right. I had to really unpack what Julia was doing on several occasions as it was actually somewhat ambiguous syntactically.

File: src/auspex/analysis/fits.py (hunk 1 above)
Comment (grahamrow) [reply src/auspex/analysis/fits.py]: The scipy version doesn't take complex inputs, which are used for the cleaned data.

File: src/auspex/analysis/fits.py (hunk 2 above)
Comment (grahamrow) [reply src/auspex/analysis/fits.py]: True.

File: src/auspex/analysis/fits.py (hunk 5 above)
Comment (grahamrow) [reply src/auspex/analysis/fits.py]: as long as the function names are snake_cased (which is arguable) I'll give it a pass.

File: src/auspex/analysis/fits.py (hunk 3 above)
Comment (grahamrow) [reply src/auspex/analysis/fits.py]: Good suggestion.

File: src/auspex/analysis/fits.py (hunk 1 above)
Comment (caryan) [reply src/auspex/analysis/fits.py]: true but I think it goes right through and is easier than writing your own:

```
cleaned_analytical_sig = hilbert(np.real(cleaned_data)) +1j * hilbert(np.imag(cleaned_data))
```

**Pull request merged**